pkg/controller/datavolume/controller-base.go

```diff
@@ -1125,8 +1156,35 @@ func (r *ReconcilerBase) storageClassCSIDriverExists(storageClassName *string) (
 	csiDriver := &storagev1.CSIDriver{}
 
 	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: storageClass.Provisioner}, csiDriver); err != nil {
-		return false, err
+		if !k8serrors.IsNotFound(err) {
+			return false, err
+		}
+		return false, nil
 	}
 
 	return true, nil
 }
+
+func (r *ReconcilerBase) checkPVCUsingPopulators(pvc *corev1.PersistentVolumeClaim) (bool, error) {
+	if pvc.Spec.DataSourceRef == nil {
+		return false, nil
+	}
+	return r.storageClassCSIDriverExists(pvc.Spec.StorageClassName)
```

Comment:
rather than look up the driver again, is there an annotation on the pvc we can safely rely on?